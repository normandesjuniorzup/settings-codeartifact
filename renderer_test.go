package main_test

import (
	"bytes"
	"io"
	. "settings"
	"testing"

	approvals "github.com/approvals/go-approval-tests"
)

func TestRender(t *testing.T) {
	buf := bytes.Buffer{}
	token := "some-token"

	settingsRenderer, err := NewSettingsRenderer()
	if err != nil {
		t.Fatal(err)
	}

	if err := settingsRenderer.Render(&buf, token); err != nil {
		t.Fatal(err)
	}

	approvals.VerifyString(t, buf.String())
}

func BenchmarkRender(b *testing.B) {
	token := "bench-token"

	settingsRenderer, err := NewSettingsRenderer()
	if err != nil {
		b.Fatal(err)
	}

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		settingsRenderer.Render(io.Discard, token)
	}
}
