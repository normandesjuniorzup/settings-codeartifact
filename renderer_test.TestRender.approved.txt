<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xsi="http://www.w3.org/2001/XMLSchema-instance"
  schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
https://maven.apache.org/xsd/settings-1.0.0.xsd">

  <localRepository>${user.home}/.m2/repository</localRepository>
  <interactiveMode>true</interactiveMode>
  <usePluginRegistry>false</usePluginRegistry>
  <offline>false</offline>

  <servers>
    <server>
      <id>claro-codeartifact-claro-flex-artifacts</id>
      <username>aws</username>
      <password>some-token</password>
    </server>
  </servers>

  <profiles>
    <profile>
      <id>claro-codeartifact-claro-flex-artifacts</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <repositories>
        <repository>
          <id>claro-codeartifact-claro-flex-artifacts</id>
          <url>
            https://claro-codeartifact-319569500149.d.codeartifact.us-east-1.amazonaws.com/maven/claro-flex-artifacts/</url>
          <releases>
            <checksumPolicy>fail</checksumPolicy>
          </releases>
        </repository>
      </repositories>
    </profile>
  </profiles>

  <mirrors>
    <mirror>
      <id>claro-codeartifact-claro-flex-artifacts</id>
      <name>claro-codeartifact-claro-flex-artifacts</name>
      <url>
        https://claro-codeartifact-319569500149.d.codeartifact.us-east-1.amazonaws.com/maven/claro-flex-artifacts/</url>
      <mirrorOf>*</mirrorOf>
    </mirror>
  </mirrors>

</settings>